File: 0289-game-of-life/0289-game-of-life.py
class Solution:
    def gameOfLife(self, board: List[List[int]]) -> None:
        m,n=len(board),len(board[0])
        dx=[-1,-1,-1,0,0,1,1,1]
        dy=[-1,0,1,-1,1,-1,0,1]
        for i in range(m):
            for j in range(n):
                live=0
                for d in range(8):
                    ni,nj=i+dx[d],j+dy[d]
                    if 0<=ni<m and 0<=nj<n and abs(board[ni][nj])==1:
                        live+=1
                if board[i][j]==1 and (live<2 or live>3):
                    board[i][j]=-1
                if board[i][j]==0 and live==3:
                    board[i][j]=2
        for i in range(m):
            for j in range(n):
                if board[i][j]==-1:
                    board[i][j]=0
                if board[i][j]==2:
                    board[i][j]=1

File: 0289-game-of-life/0289-game-of-life.c
void gameOfLife(int** board, int boardSize, int* boardColSize) {
    int m=boardSize,n=boardColSize[0];
    int dx[8]={-1,-1,-1,0,0,1,1,1};
    int dy[8]={-1,0,1,-1,1,-1,0,1};
    for(int i=0;i<m;i++){
        for(int j=0;j<n;j++){
            int live=0;
            for(int d=0;d<8;d++){
                int ni=i+dx[d],nj=j+dy[d];
                if(ni>=0&&ni<m&&nj>=0&&nj<n&&abs(board[ni][nj])==1){
                    live++;
                }
            }
            if(board[i][j]==1&&(live<2||live>3)){
                board[i][j]=-1;
            }
            if(board[i][j]==0&&live==3){
                board[i][j]=2;
            }
        }
    }
    for(int i=0;i<m;i++){
        for(int j=0;j<n;j++){
            if(board[i][j]==-1) board[i][j]=0;
            if(board[i][j]==2) board[i][j]=1;
        }
    }
}